ferretdb/runner.go

```diff
@@ -0,0 +1,131 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package provides support for the embedded use-case.
+package ferretdb
+
+import (
+	"context"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+
+	"github.com/FerretDB/FerretDB/internal/clientconn"
+	"github.com/FerretDB/FerretDB/internal/handlers"
+	"github.com/FerretDB/FerretDB/internal/util/debug"
+	"github.com/FerretDB/FerretDB/internal/util/logging"
+	"github.com/FerretDB/FerretDB/internal/util/must"
+	"github.com/FerretDB/FerretDB/internal/util/version"
+)
+
+// config is a FerretDB library config.
+var config Config
+
+// Config ConnectionString contains a backend connection string.
+// "postgres://user@postgres:5432/ferretdb" - then it's postgres.
+type Config struct {
+	ConnectionString string
+}
+
+// registeredHandlers maps handler names to constructors.
+// The values for `registeredHandlers` must be set through the `init()` functions of the corresponding handlers
+// so that we can control which handlers will be included in the build with build tags.
+var registeredHandlers = map[string]newHandler{}
+
+// newHandler represents a function that constructs a new handler.
+type newHandler func(opts *newHandlerOpts) (handlers.Interface, error)
+
+// newHandlerOpts represents common configuration for constructing handlers.
+//
+// Handler-specific configuration is not modifiable in the embedded use-case.
+type newHandlerOpts struct {
+	ctx    context.Context
+	logger *zap.Logger
+}
+
+// run function that runs FerretDB in NormalMode until ctx is canceled.
+func run(ctx context.Context, conf Config) error {
```

Comment:
There is no way to use that unexported function

Reply:
> There is no way to use that unexported function

Please give more information on your thoughts.

Reply:
removed.